gneiss/plot/_regression_plot.py

```diff
@@ -0,0 +1,369 @@
+    X = pd.Series(np.arange(len(pvals.index)), index=pvals.index)
+    Y = pd.Series(np.arange(len(pvals.columns)), index=pvals.columns)
+    m['X'] = [X.loc[i] for i in m.balance]
+    m['Y'] = [Y.loc[i] for i in m.Covariate]
+
+    # fill in nans with zero.  Sometimes the pvalue calculation fails.
+    m = m.fillna(0)
+    for i in m.index:
+        x = m.loc[i, 'log_Pvalue']
+
+        ind = int(np.floor((x - _min) / (_max - _min) * (N - 1)))
+        m.loc[i, 'color'] = palette[ind]
+
+    source = ColumnDataSource(ColumnDataSource.from_df(m))
+    hm = figure(title='Regression Coefficients Summary',
+                plot_width=1200, plot_height=400,
+                tools=[hover, PanTool(), BoxZoomTool(),
+                       WheelZoomTool(), ResetTool(),
+                       SaveTool()])
+    hm.rect(x='X', y='Y', width=1, height=1,
+            fill_color='color', line_color="white", source=source)
+    Xlabels = pd.Series(pvals.index, index=np.arange(len(pvals.index)))
+    Ylabels = pd.Series(pvals.columns, index=np.arange(len(pvals.columns)), )
+
+    hm.xaxis[0].ticker = FixedTicker(ticks=Xlabels.index)
+    hm.xaxis.formatter = FuncTickFormatter(code="""
+    var labels = %s;
+    return labels[tick];
+    """ % Xlabels.to_dict())
+
+    hm.yaxis[0].ticker = FixedTicker(ticks=Ylabels.index)
+    hm.yaxis.formatter = FuncTickFormatter(code="""
+    var labels = %s;
+    return labels[tick];
+    """ % Ylabels.to_dict())
+
+    return hm
+
+
+def _decorate_tree(t, series):
+    """ Attaches some default values on the tree for plotting.
+
+    Parameters
+    ----------
+    t: skbio.TreeNode
+        Input tree
+    series: pd.Series
+        Input pandas series
+
+    """
+    for i, n in enumerate(t.postorder()):
+        n.size = 30
+        if n.is_root():
+            n.size = 50
+        elif n.name == n.parent.children[0].name:
+            n.color = '#00FF00'  # left child is green
+        else:
+            n.color = '#FF0000'  # right child is red
+        if not n.is_tip():
+            t.length = series.loc[n.name]
+    return t
+
+
+def _deposit_results(model, output_dir):
+    """ Store all of the core regression results into a folder. """
+    coefficients = model.coefficients()
+    coefficients.to_csv(os.path.join(output_dir, 'coefficients.csv'),
+                        header=True, index=True)
+    residuals = model.residuals()
+    residuals.to_csv(os.path.join(output_dir, 'residuals.csv'),
+                     header=True, index=True)
+    predicted = model.predict()
+    predicted.to_csv(os.path.join(output_dir, 'predicted.csv'),
+                     header=True, index=True)
+    pvalues = model.pvalues
+    pvalues.to_csv(os.path.join(output_dir, 'pvalues.csv'),
+                   header=True, index=True)
+    balances = model.balances
+    balances.to_csv(os.path.join(output_dir, 'balances.csv'),
+                    header=True, index=True)
+
+
+def _deposit_results_html(index_f):
```

Comment:
This name is confusing with _deposit_results, as this one write some text to create some link and _deposit_results writes the results to disk, perhaps renaming will be worth it: generate_html_links?